### ISSUE TITLE
Card Layout in column breaks site width

### ISSUE THREAD

**Issue by HashandSalt** [getkirby/kirby#1666]
**Describe the bug**
Setting a file sections with large card layout enabled, inside a 1/3 right hand column breaks the width of the panel.

**To Reproduce**
1. Set two columns in a blue print, left 2/3, right 1/3
2. Set a files section in the right  hand column
3. Set layout to card, ratio to 16/9

My files section for reference:

```
heroimagefiles:
  type:                   files
  template:               imagefocus
  headline:               Images
  layout:                 cards
  size:                   large
  image:
    cover:                true
    ratio:                16/9
    back:                 black
```

**Expected behavior**
Card fills the column width.

**Screenshots**

![card-bug](https://user-images.githubusercontent.com/28439755/55745110-6621d700-5a2e-11e9-85ad-34a828598ef4.jpg)

**Kirby Version**
Kirby 3.1.1

**Desktop**
 - OS: macOS High Sierra
 - Opera Browser 58.0.3135.127

**Comment by afbora** [comment]
@distantnative I have few tests on 3.2.5 and this issue about following lines. When I deleted these lines, it fixed in all screen resolutions. To be honest, I don't know exactly what it does and why here 😅 

````css
@media screen and (min-width: $breakpoint-medium) {
  .k-cards[data-size="large"] {
    grid-template-columns: repeat(auto-fill, minmax(32rem, 1fr));
  }
}
````

https://github.com/getkirby/kirby/blob/develop/panel/src/components/Layout/Cards.vue#L45-L49

**Comment by HashandSalt** [comment]
@afbora It is probably the `32rem` part doing it. That rule basically makes the cards fit equally, but they won't get ever get a width of less than 32rem. If the panels base font size is 12px, then 32rem is 32x that = 384px. It should be set to whatever the smallest possible column width is at the least. 

Ideally though, it shouldn't kick in unless the card is in a column that's wide enough to handle multiple cards. Therefore, it probably needs a parent selector before the `.k-cards` so that it can only happen on 2/3, 3/3, 1/2, 2/2, 2/4. 3/4, 4/4 columns because 1/3 and 1/4 columns are probably too small. 

Basically the rule needs to kick in if the screen size is large enough AND the parent column is large enough to cope with more than one card side by side, and the min card width needs to be no greater than the smallest a 1/4 column can get.

**Comment by distantnative** [comment]
> Basically the rule needs to kick in if the screen size is large enough AND the parent column is large enough to cope with more than one card side by side, and the min card width needs to be no greater than the smallest a 1/4 column can get.

Indeed. And I think without CSS container queries, this won't be solvable.

**Comment by distantnative** [comment]
I just got an idea from:
https://shayneo.com/blog/binding-css-variables-with-vue/

Using computed props with the style attribute to actually define styles based on the parent container.

**Comment by distantnative** [comment]
Basically, @HashandSalt described the exact problem and solution. But I only figured out today how to actually do that in CSS with an extra `min()`: https://github.com/getkirby/kirby/pull/2494

**Comment by bastianallgeier** [comment]
✅
